src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -184,6 +205,26 @@ public void onRunFinish(boolean succeeded, BatchSourceContext context) {
     } catch (IOException e) {
       LOG.warn("Failed to delete bucket " + gcsPath.toUri().getPath() + ", " + e.getMessage());
     }
+    if (configuration.get(BigQueryConstants.CONFIG_PARTITION_FROM_DATE) != null
+      || configuration.get(BigQueryConstants.CONFIG_PARTITION_TO_DATE) != null) {
+      try {
+        BigQueryFactory factory = new BigQueryFactory();
+        BigQueryHelper bigQueryHelper = factory.getBigQueryHelper(configuration);
+        Bigquery.Tables tables = bigQueryHelper.getRawBigquery().tables();
+        Bigquery.Tables.Delete delete = tables.delete(
```

Comment:
BigQuery has temporary tables that get cleaned up automatically after some expiry (BQ queries actually get stored in temporary tables). We should be relying on that functionality, as it's possible this logic never gets to run if the pipeline is killed.

Reply:
Logic was changed. Currently temporary table is created with expiration time 1 day. Automatic deleting of table was removed.